rolling-update-kubernetes/src/main/scala/akka/rollingupdate/kubernetes/PodDeletionCost.scala

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (C) 2017-2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.rollingupdate.kubernetes
+
+import akka.actor.ActorSystem
+import akka.actor.ClassicActorSystemProvider
+import akka.actor.ExtendedActorSystem
+import akka.actor.Extension
+import akka.actor.ExtensionId
+import akka.actor.ExtensionIdProvider
+import akka.actor.Props
+import akka.annotation.InternalApi
+import akka.dispatch.MessageDispatcher
+import akka.event.Logging
+import akka.rollingupdate.kubernetes.PodDeletionCost.Internal.BootstrapStep
+import akka.rollingupdate.kubernetes.PodDeletionCost.Internal.Initializing
+import akka.rollingupdate.kubernetes.PodDeletionCost.Internal.NotRunning
+
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.util.concurrent.atomic.AtomicReference
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+final class PodDeletionCost(implicit system: ExtendedActorSystem) extends Extension {
+
+  private val log = Logging(system, classOf[PodDeletionCost])
+  private val configPath = "akka.rollingupdate.kubernetes"
+  private val config = system.settings.config.getConfig(configPath)
+  private val k8sSettings = KubernetesSettings(config)
+  private val costSettings = PodDeletionCostSettings(config)
+  log.debug("Settings {}", k8sSettings)
+
+  private final val startStep = new AtomicReference[BootstrapStep](NotRunning)
+
+  def start(): Unit = {
+    if (k8sSettings.podName.isEmpty) {
+      log.warning(
+        "No configuration found to extract the pod name from. " +
+        s"Be sure to provide the pod name with `$configPath.pod-name` " +
+        "or by setting ENV variable `KUBERNETES_POD_NAME`.")
+    } else if (startStep.compareAndSet(NotRunning, Initializing)) {
+      log.debug("Starting PodDeletionCost for podName={} with settings={}", k8sSettings.podName, costSettings)
+
+      implicit val blockingDispatcher: MessageDispatcher = system.dispatchers.lookup("blocking-io-dispatcher")
```

Comment:
```suggestion
      implicit val blockingDispatcher: ExecutionContext = system.dispatchers.lookup("blocking-io-dispatcher")
```
Is what we need it for here